components/footer/src/index.js

```diff
@@ -0,0 +1,158 @@
+    color: `${FOOTER_TEXT}`,
+  },
+  typography.font({ size: 16 }),
+  spacing.withWhiteSpace({ padding: { size: 7, direction: 'top' } }),
+  spacing.withWhiteSpace({ padding: { size: 5, direction: 'bottom' } })
+);
+
+/**
+ *
+ * ### Usage
+ *
+ * Simple
+ * ```jsx
+ * <Footer />
+ * ```
+ *
+ * Footer with Copyright information
+ * ```jsx
+ * // Import self hosted copyright image
+ * import crest from './govuk-crest.png';
+ *
+ * <Footer
+ *   copyright={{
+ *     text: 'Crown copyright',
+ *     link:
+ *       'https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/',
+ *     image: {
+ *       source: crest,
+ *       height: 102,
+ *       width: 125,
+ *     },
+ *   }}
+ * />;
+ * ```
+ *
+ * Footer with navigation.  There is also the option to provide footer links with, or without React/Reach Router.
+ * ```jsx
+ * import { Link } from 'react-router-dom';
+ *
+ * <Footer>
+ *   <Footer.Navigation>
+ *     <Footer.NavigationLinks heading="Two column list" listColumns="2">
+ *       <Footer.Link href="/">Navigation item 1</Footer.Link>
+ *       <Footer.Link to="/footer-nav-item-2" as={Link}>Navigation item 2 (Router Link)</Footer.Link>
+ *       <Footer.Link href="/">Navigation item 3</Footer.Link>
+ *       <Footer.Link href="/">Navigation item 4</Footer.Link>
+ *       <Footer.Link href="/">Navigation item 5</Footer.Link>
+ *       <Footer.Link href="/">Navigation item 6</Footer.Link>
+ *     </Footer.NavigationLinks>
+ *     <Footer.NavigationLinks heading="Single column list">
+ *       <Footer.Link href="/">Navigation item 1</Footer.Link>
+ *       <Footer.Link href="/">Navigation item 2</Footer.Link>
+ *       <Footer.Link href="/">Navigation item 3</Footer.Link>
+ *     </Footer.NavigationLinks>
+ *   </Footer.Navigation>
+ * </Footer>
+ * ```
+ *
+ * Footer with links in meta area
+ * ```jsx
+ * <Footer meta={
+ *     <Footer.MetaLinks heading="Support links">
+ *       <Footer.Link href="/">Item 1</Footer.Link>
+ *       <Footer.Link to="/footer-meta-item-2" as={Link}>Item 2 (Router Link)</Footer.Link>
+ *       <Footer.Link href="/">Item 3</Footer.Link>
+ *     </Footer.MetaLinks>
+ *   }
+ * />
+ * ```
+ *
+ * Footer with custom content in meta area
+ * ```jsx
+ * <Footer meta={
+ *     <Footer.MetaCustom>
+ *       Built by the <Footer.Link href="/">Government Digital Service</Footer.Link>
+ *     </Footer.MetaCustom>
+ *   }
+ * />
+ * ```
+ *
+ * NB <Footer.Link /> is styled extention of @govuk-react/link
+ *
+ * ### References:
+ * - https://design-system.service.gov.uk/components/footer/
+ * - https://github.com/alphagov/govuk-frontend/blob/master/src/components/footer/template.njk
+ * - https://github.com/alphagov/govuk-frontend/blob/master/src/components/footer/_footer.scss
+ *
+ */
+const Footer = ({ children, copyright, meta }) => (
+  <FooterContainer role="contentinfo">
+    <WidthContainer>
```

Comment:
It may be sensible to make `WidthContainer` be prop-driven, and thus allow for it to be overridden/customised.

For reference, see the `page` component and it's storybook examples that demonstrate how this can work.  There's an example on `page` that allows for the page to be formatted beyond the standard max width - it would be nice if the footer could match that.